.circleci/config.yml

```diff
@@ -30,68 +27,62 @@ aliases:
             poetry run ./.circleci/add-known-hosts.sh
             chmod +x .circleci/github-pages.sh
             poetry run ./.circleci/github-pages.sh
+          environment:
+            REPOSITORY_PATH: /home/circleci/docs
+            DEPLOYMENT_GIT_SSH: git@github.com:Sceptre/sceptre.github.io.git
 
 jobs:
   build:
-    docker:
-      - image: sceptreorg/sceptre-circleci:2.1.0
+    executor: python/default
     steps:
       - checkout
       - restore_cache:
-          key: sceptre-v1-{{ checksum "pyproject.toml" }}-{{ checksum "poetry.lock" }}-{{ checksum ".pre-commit-config.yaml" }}-{{ checksum "tox.ini" }}
-      - run:
-          name: 'Installing Dependencies'
-          command: poetry install --all-extras -v
+          key: sceptre-v1-{{ checksum ".pre-commit-config.yaml" }}
+      - python/install-packages:
+          pkg-manager: poetry
+          args: --all-extras
       - run:
           name: Run pre-commit
           command: poetry run pre-commit run --all-files --show-diff-on-failure
       - run:
           name: Build package
           command: poetry build
       - save_cache:
-          key: sceptre-v1-{{ checksum "pyproject.toml" }}-{{ checksum "poetry.lock" }}-{{ checksum ".pre-commit-config.yaml" }}-{{ checksum "tox.ini" }}
+          key: sceptre-v1-{{ checksum ".pre-commit-config.yaml" }}
           paths:
-            - '~/.cache/pypoetry'
-            - '~/.cache/pre-commit'
-            - '.tox'
+            - ~/.cache/pre-commit
 
   unit-tests:
-    docker:
-      - image: sceptreorg/sceptre-circleci:2.1.0
+    parameters:
+      python:
+        type: string
+    executor:
+      name: python/default
+      tag: <<parameters.python>>
     steps:
       - checkout
-      - restore_cache:
-          key: sceptre-v1-{{ checksum "pyproject.toml" }}-{{ checksum "poetry.lock" }}-{{ checksum ".pre-commit-config.yaml" }}-{{ checksum "tox.ini" }}
+      - python/install-packages:
+          pkg-manager: poetry
+          args: --all-extras
       - run:
           name: Unit Tests
           command: |
-            poetry run tox -e clean
-            poetry run tox
-            poetry run tox -e report
+            poetry run pytest --cov=sceptre --cov-report=term-missing --cov-report=html --junitxml=test-reports/junit.xml
```

Comment:
now that the CI is no longer running with tox maybe we should change [skip_missing_interpreters](https://github.com/Sceptre/sceptre/blob/master/tox.ini#L4) to `true` ?

Reply:
Yes, I agree. Tox is still useful for local unit testing and the skip action makes it easier to use.

I wonder now whether we even need tox for local testing. Can you run a CircleCI test confirguration locally?

Reply:
updated `skip_missing_interpreters`

> Can you run a CircleCI test confirguration locally?

With Docker [yes](https://circleci.com/docs/how-to-use-the-circleci-local-cli/#run-a-job-in-a-container-on-your-machine), but imo tox is still easier/lighterweight for local dev

Reply:
actually, how about if we just run unit tests with tox, `poetry run tox`?  with `skip_missing_interpreters` enabled tox should only run tests on available python versions.  This should have the same affect as pytest except that we would not be ignoring the tox python versions list.

Reply:
@branchvincent , why do you want to use `pytest` here instead of `tox`? I thought it was fine to just use `tox` here with `skip_missing_interpreters=true` because of what @zaro0508 said.

Reply:
(sorry just getting back to this)

I think I did this since the container has multiple pythons (one as expected matching the image tag but also an OS-provided one):

```console
$ docker run --rm -it cimg/python:3.12
$ which -a python3
/home/circleci/.pyenv/shims/python3
/usr/bin/python3
/bin/python3
$ which -a python3 | xargs -I{} echo {} -V | sh
Python 3.12.1
Python 3.10.6
Python 3.10.6
```

So just `poetry run tox` would run tests for both versions (which isn't ideal since we may see a failing `ci/circleci: unit-tests-3.12` job and learn that actually it was a failure for 3.10). So we'd want `poetry run tox -e py312`, let me see what that would look like

Reply:
updated to use `tox`